spec.emu

```diff
@@ -56,4 +56,44 @@ location: https://tc39.es/proposal-array-filtering/
     </emu-note>
   </emu-clause>
   </ins>
+
+  <ins class="block">
+  <emu-clause id="sec-array.prototype.groupby">
+    <h1>Array.prototype.groupBy ( _callbackfn_ [ , _thisArg_ ] )</h1>
+    <emu-note>
+      <p>_callbackfn_ should be a function that accepts three arguments and returns a value that is coercible to a property. `groupBy` calls _callbackfn_ once for each element in the array, in ascending order, and constructs a new object of arrays, where the property key is the return value and array is filled with the items that return that property key. _callbackfn_ is called only for elements of the array which actually exist; it is not called for missing elements of the array.</p>
+      <p>If a _thisArg_ parameter is provided, it will be used as the *this* value for each invocation of _callbackfn_. If it is not provided, *undefined* is used instead.</p>
+      <p>_callbackfn_ is called with three arguments: the value of the element, the index of the element, and the object being traversed.</p>
+      <p>`groupBy` does not directly mutate the object on which it is called but the object may be mutated by the calls to _callbackfn_.</p>
+      <p>The range of elements processed by `groupBy` is set before the first call to _callbackfn_. Elements which are appended to the array after the call to `groupBy` begins will not be visited by _callbackfn_. If existing elements of the array are changed their value as passed to _callbackfn_ will be the value at the time `groupBy` visits them; elements that are deleted after the call to `groupBy` begins and before being visited are not visited.</p>
+    </emu-note>
+    <p>When the `groupBy` method is called with one or two arguments, the following steps are taken:</p>
+    <emu-alg>
+      1. Let _O_ be ? ToObject(*this* value).
+      1. Let _len_ be ? LengthOfArrayLike(_O_).
+      1. If IsCallable(_callbackfn_) is *false*, throw a *TypeError* exception.
+      1. Let _k_ be 0.
+      1. Let _obj_ be ! OrdinaryObjectCreate(%Object.prototype%).
```

Comment:
Maybe `null` prototype could be better?

Reply:
Quite possibly, but this would need to be discussed in plenary. I'll add it to the slides, thanks!

Reply:
The spec typically doesn't create null objects; module namespace objects and Object.create(null) are outliers. As long as users use Object.keys/entries/values/assign/object spread with these objects, or direct property access of expected groups (and thus, present as own properties) - all of which are what I'd expect the entirety of use cases to be - it wouldn't make a difference if it had a null proto or not.

Reply:
In some cases it's used in the spec - for example, `RegExp` named groups - and it's a similar case.

Users will use it
```js
const { foo = [], bar = [], baz = [] } = array.groupBy(fn);
```

It makes no sense to force users to check `Object.hasOwn` each time for preventing
```js
Object.prototype.foo = ['oops'];
```

Reply:
Why would they need to? They’ll know which properties they’ve just grouped by.

Reply:
Some groups can haven't any elements -> missed on the result.